File: src/commands/current.ts
// muxx current — print the name of the currently attached session

import { isInsideTmux } from "../core/env.js";
import { currentSession } from "../core/tmux.js";

export async function current(): Promise<void> {
  if (!isInsideTmux()) {
    console.error("not inside a tmux session");
    process.exit(1);
  }

  const name = currentSession();
  if (!name) {
    console.error("could not determine current session");
    process.exit(1);
  }

  console.log(name);
}

File: src/commands/current.rs
use anyhow::{bail, Result};

use crate::core::{env::is_inside_tmux, tmux::current_session};

pub fn run() -> Result<()> {
    if !is_inside_tmux() {
        bail!("not inside a tmux session");
    }

    match current_session() {
        Some(name) => {
            println!("{name}");
            Ok(())
        }
        None => bail!("could not determine current session"),
    }
}
